SignalServiceKit/src/Profiles/OWSProfilesManager.m

```diff
@@ -8,11 +8,79 @@
 #import "TSStorageManager.h"
 #import "TextSecureKitEnv.h"
 
+#import "TSYapDatabaseObject.h"
+
 NS_ASSUME_NONNULL_BEGIN
 
+@class TSThread;
+
+@interface AvatarMetadata : TSYapDatabaseObject
+
+// This filename is relative to OWSProfilesManager.profileAvatarsDirPath.
+@property (nonatomic, readonly) NSString *fileName;
+@property (nonatomic, readonly) NSString *avatarUrl;
+@property (nonatomic, readonly) NSString *avatarDigest;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+@end
+
+#pragma mark -
+
+@implementation AvatarMetadata
+
++ (NSString *)collection
+{
+    return @"AvatarMetadata";
+}
+
+- (instancetype)initWithFileName:(NSString *)fileName
+                       avatarUrl:(NSString *)avatarUrl
+                    avatarDigest:(NSString *)avatarDigest
+{
+    // TODO: Local filenames for avatars are guaranteed to be unique.
+    self = [super initWithUniqueId:fileName];
+
+    if (!self) {
+        return self;
+    }
+
+    OWSAssert(fileName.length > 0);
+    OWSAssert(avatarUrl.length > 0);
+    OWSAssert(avatarDigest.length > 0);
+    _fileName = fileName;
+    _avatarUrl = avatarUrl;
+    _avatarDigest = avatarDigest;
+
+    return self;
+}
+
+
+#pragma mark - NSObject
+
+- (BOOL)isEqual:(AvatarMetadata *)other
+{
+    return ([other isKindOfClass:[AvatarMetadata class]] && [self.fileName isEqualToString:other.fileName] &&
+        [self.avatarUrl isEqualToString:other.avatarUrl] && [self.avatarDigest isEqualToString:other.avatarDigest]);
+}
+
+- (NSUInteger)hash
+{
+    return self.fileName.hash ^ self.avatarUrl.hash ^ self.avatarDigest.hash;
+}
+
+@end
+
+#pragma mark -
+
+NSString *const kNSNotificationName_LocalProfileDidChange = @"kNSNotificationName_LocalProfileDidChange";
+
 NSString *const kOWSProfilesManager_Collection = @"kOWSProfilesManager_Collection";
 // This key is used to persist the local user's profile key.
 NSString *const kOWSProfilesManager_LocalProfileKey = @"kOWSProfilesManager_LocalProfileKey";
```

Comment:
I got a little confused because we're overloading the word "key" here. Rather than call it the `LocalProfileKeyKey`, maybe we could call it the `LocalProfileSecretKey`